kernel/src/main/scala/me/rexim/morganey/ast/LambdaTerm.scala

```diff
@@ -46,6 +47,30 @@ sealed trait LambdaTerm extends MorganeyNode {
       case (MorganeyBinding(variable, value), acc) =>
         LambdaApp(LambdaFunc(variable, acc), value)
     }
+
+  def addDependentBindings(context: Seq[MorganeyBinding],
+                           currentVars: Set[String] = Set()): Either[String, LambdaTerm] = {
+    val conflictedVars = freeVars & currentVars
+
+    if (conflictedVars.isEmpty) {
+      val bindings: List[Either[String, MorganeyBinding]] = freeVars.toList.map { x =>
+        context
+          .find(_.variable.name == x)
+          .toRight(s"Unexisting binding: $x")
```

Comment:
If you meant "a binding that doesn't exist", then you should've used "non-existent", not "unexisting".